### ISSUE TITLE
Gradient checkpointing and Gradient accumulation Support

### ISSUE THREAD

**Issue by dathudeptrai** [keras-team/keras-nlp#1152]
Training LLM models on small GPUs is a very common use-case. Gradient checkpointing and Gradient Accumulation are two dominant techniques nowadays. As we know that the `XLA` worked very well with GA (https://www.tensorflow.org/xla) so I just wonder if it is indispensable to add these 2 techniques to this repo and AFAIK, there is no good opensources out there.
